evm/p2p/protocol.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import struct
+
+import rlp
+from rlp import sedes
+
+from evm.constants import NULL_BYTE
+
+
+class Command:
+    id = None
+    decode_strict = True
+    structure = []
+
+    @classmethod
+    def encode_payload(cls, data):
+        if isinstance(data, dict):  # convert dict to ordered list
+            if not isinstance(cls.structure, list):
+                raise ValueError("Command.structure must be a list when data is a dict")
+            data = [data[name] for (name, _) in cls.structure]
```

Comment:
Feels weird that if `data` had a key in it that isn't in `cls.structure` that it would get silently dropped.  Unless there is a good reason to allow this, can you add validation that the the keys match?

Reply:
No reason for that, added validation